### ISSUE TITLE
Create SAC object in database given validated user input from the frontend

### ISSUE THREAD

**Issue by petermarks-gsa** [GSA-TTS/FAC#168]
Background: In https://github.com/GSA-TTS/FAC/pull/152 we implemented a pathway to create an SF-SAC instance from the combined API calls of a user navigating through the form intake steps.

Depends on #128 

Goal is to fine-turn the implementing from https://github.com/GSA-TTS/FAC/pull/152 to meet the needs of the frontend as developed as part of https://github.com/GSA-TTS/FAC/issues/128

Open questions:
* Does current model account for notion of a complete / incomplete state. Joe - Yes
* What are other validation rules? Joe - validation similar to UEI model. Once they have completed final intake step, how do we take sum total of data and write it to the DB as an instance of this model. 

## We'll know we're done when
- [ ] Upon successful completion of the Access step, the backend creates a `SingleAuditChecklist` instance with the user's provided data.

**Comment by petermarks-gsa** [comment]
@Jkrzy to relay questions in slack into here. This might be slightly bumped down the user flow.

**Comment by Jkrzy** [comment]
In #152 we implemented a pathway to create an SF-SAC instance from the combined API calls of a user navigating through the form intake steps.

I'd like to move this issue to be a task of #128, focused on fine tuning the implementing from #152 to meet the needs of the frontend as developed as part of #128

**Comment by tadhg-ohiggins** [comment]
While working on this issue Joe and I discussed the possible ways to handle form status; I'm including some of the 

## Some definitions

<dl>
<dt>Intake form data</dt>
<dd>Preliminary data about the audit submission, including UEI, email addresses of users, year of submission</dd>
<dt>SAC full validation</dt>
<dd>The SAC model in our system must allow incomplete/invalid submissions so that in-progress forms can be saved in our backend. This means we need another step, using application code that is not part of Django model validation, to verify that a form is valid and complete.</dd>
</dl>

## Possible SAC statuses

Below are the statuses in sequential order.

<dl>
<dt>Created/Submitted</dt>
<dd>The intake form data was sufficient to allow the user to begin the SAC process.</dd>
<dt>In Progress</dt>
<dd>The user(s) have entered some data post-creation.</dd>
<dt>Certifiable</dt>
<dd>The form is valid (complete and with valid entries) according to our SAC full validation step.</dd>
<dt>Auditee Certified</dt>
<dd>The form is valid (complete and with valid entries) according to our SAC full validation step and the auditee has marked it as certified.</dd>
<dt>Certified</dt>
<dd>The form is valid (complete and with valid entries) according to our SAC full validation step and the auditee and auditor have both marked it as certified.</dd>
<dt>Awaiting Manual Verification</dt>
<dd>A human examines the overall submission and checks to see that for example the submitted PDF is what it should be. Final quality control check before the FAC formally accepts the submission.</dd>
<dt>Accepted</dt>
<dd>FAC has accepted the submission and the entity's regulatory obligation to submit audit data has been fulfilled.</dd>
</dl>

## Related notes

We need to track the time of any status change (as well as who made the last change before that).

We think we shouldn't make attestation controls available if the form is incomplete/invalid.

Consider using Django reversion module (see #199).

-----

This issue, #168, is about getting from having sufficient intake form data to the Created/Submitted status for SAC.

I will create a ``status`` field for the ``SingleAuditChecklist`` model and possibly an enumeration of the statuses, but no logics related to status changes is part of this issue.

Issue #152 is probably the most relevant prior PR to this issue.

**Comment by petermarks-gsa** [comment]
@tadhg-ohiggins @Jkrzy Thanks for the thought you put into this. Looping in this story's "shepherd" @MelissaBraxton to discuss. I have a couple questions:

1. Regarding the "Possible SAC statuses" list, I'm not aware of an equivalent status in the current FAC. Is that the case? 

2. In the current data model, I see various date fields which seem to in effect capture the status milestone at a certain point:

AUDITEEDATESIGNED
CPADATESIGNED
INITIAL DATE RECEIVED
FORM DATE RECEIVED
COMPONENT DATE RECEIVED
FAC ACCEPTED DATE

Per my [Rosetta Stone document](https://docs.google.com/spreadsheets/d/1e-NQPeXk9pcQhA9PEbywkfoJa1bcTunKsVsBjqFYVK4/edit#gid=1955175057), I'm not aware of these existing in our current data model. If there's not a plan to reintroduce those, what's the reason for that and what's our plan for migrating existing data?

**Comment by tadhg-ohiggins** [comment]
Conversation about the statuses has moved to a separate issue, #202.

**Comment by tadhg-ohiggins** [comment]
While the question of what the statuses will be and the way in which a submission can move between them is still open, for this issue I'll proceed with having `status` as a property of a submission, and I'll use a set of statuses that we can change in future.

**Comment by tadhg-ohiggins** [comment]
Now that statuses are represented in the model, we can consider this issue closed.
